src/backend/distributed/executor/executor.c

```diff
@@ -1080,16 +1179,25 @@ RunDistributedExecution(DistributedExecution *execution)
 				execution->waitFlagsChanged = false;
 			}
 
+			if (MultiShardConnectionType == SEQUENTIAL_CONNECTION)
+			{
+				connectionsInUse = list_length(execution->workerList);
```

Comment:
Is there a way to avoid this if statement? It seems a bit hacky.

Reply:
I think we could consider keeping this for performance optimization (as the commit message says as well). Because, the `latch.c` has the following loop:
```C
	for (cur_event = set->events, cur_pollfd = set->pollfds;
		 cur_event < (set->events + set->nevents) &&
		 returned_events < nevents;
		 cur_event++, cur_pollfd++)
```

Where `connectionsInUse` == `nevents` and  `set->nevents` == `list_length(sessionList)` in `BuildWaitEventSet`.

So, we're instructing the loop to continue for `connectionsInUse` times. I don't have any numbers on the performance effect of this change.

If we're to remove this, we need to remove `Assert (maxWaitEventCount >= connectionCount + 2);` as well, because that doesn't hold true for sequential executions, where there might be more sessions than we need.

If it feels hacky, I can remove now. Once we're doing the benchmarks, I can check the effect of this change.

Reply:
removed the if/else